### ISSUE TITLE
triplet loss problem

### ISSUE THREAD

**Issue by xiaomingdaren123** [yoavalon/TensorFlowMnistTripletLoss#1]
First of all, thank you very much for the code you provided. There are two questions I hope you can help.
1、When I used the batch_hard_triplet_loss method to train a handwritten digital network, I found that if my network does not add a BN layer,After training for a while, the loss will be fixed at the margin  and acc (acc using KNN to compute) becomes 0, I looked at the _pairwise_distances, and found that the distance matrix has almost become 0 ? What is the reason for this, must we add bn layer?
2、I used batch_hard_triplet_loss in other classification task and divided it into 7 categories. The network uses VGG network. I found that  the network will also have the above situation. I only used the triple loss . I don't know why? In addition, my data set is very small, network training uses random input, is it necessary to perform triplet sampling input to the network? Is the final output of the network normalized to calculate the loss?
Thank you, I hope you can answer
![50964997-b5c63480-150b-11e9-8af4-e7e7ec066878](https://user-images.githubusercontent.com/33368313/51290899-260b1380-1a40-11e9-9597-cec8a240fecc.png)
![50965003-b9f25200-150b-11e9-9149-84956fa9883f](https://user-images.githubusercontent.com/33368313/51290900-273c4080-1a40-11e9-821f-fe2c189930b1.png)
Figure 1 is the training situation.I looked at the distance output obtained from the last layer of the network. When I found that acc was equal to 0, all elements of my dis matrix basically started to approach 0. I don’t know what caused it.
